### PR TITLE
Fix building aggregate projections when external aggregation is on

### LINKED ISSUE #39667: Insert into table with projections doesn't work with enabled external aggregation
**How to reproduce**
Run test `01710_projection_aggregation_in_order.sql` with setting  `max_bytes_before_external_group_by=1`.
```
cat queries/0_stateless/01710_projection_aggregation_in_order.sql | clickhouse-client -n --max_bytes_before_external_group_by=1
276078600
291519000
304558200
317518200
330478200
276078600
291519000
304558200
317518200
330478200
Received exception from server (version 22.7.1):
Code: 49. DB::Exception: Received from localhost:9000. DB::Exception: Projection cannot increase the number of rows in a block. It's a bug. (LOGICAL_ERROR)
(query: INSERT INTO agg SELECT
    1,
    toDateTime('2021-12-06 00:00:00') + number,
    number
FROM numbers(100000);)
```

I'm not sure whether it should work, maybe just always disable external aggregation while calculating projection.

### PR DESCRIPTION
**Pull request by amosbird**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix building aggregate projections when external aggregation is on. Mark as improvement because the case is rare and there exists easy workaround to fix it via changing settings. This fixes https://github.com/ClickHouse/ClickHouse/issues/39667 . 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Storages/ProjectionsDescription.cpp [hunk 1]
```diff
@@ -286,7 +286,7 @@ Block ProjectionDescription::calculate(const Block & block, ContextPtr context)
                                                                        : QueryProcessingStage::WithMergeableState})
                        .buildQueryPipeline();
     builder.resize(1);
-    builder.addTransform(std::make_shared<SquashingChunksTransform>(builder.getHeader(), block.rows(), block.bytes()));
+    builder.addTransform(std::make_shared<SquashingChunksTransform>(builder.getHeader(), block.rows(), 0));
```
Comment (nickitat) [inline src/Storages/ProjectionsDescription.cpp:289]: so, what was the problem?)

**Review by nickitat — APPROVED** [review]

File: src/Storages/ProjectionsDescription.cpp (hunk 1 above)
Comment (amosbird) [reply src/Storages/ProjectionsDescription.cpp:289]: The original code is actually wrong (aggregated block might contain more bytes). However, without external grouping, there will always be one block, which never triggers the bug)

Comment (azat) [reply src/Storages/ProjectionsDescription.cpp:289]: @amosbird can you put a comment? This does not looks obvious at first glance.

Comment (amosbird) [reply src/Storages/ProjectionsDescription.cpp:289]:
```suggestion
    // Generate aggregated blocks with rows less or equal than the original block.
    // There should be only one output block after this transformation.
    builder.addTransform(std::make_shared<SquashingChunksTransform>(builder.getHeader(), block.rows(), 0));
```

**Comment by nickitat** [comment]
let's rerun tests

**Comment by nickitat** [comment]
Stress test (undefined) #39857

**Pull request merged**